packages/backend-core/src/context/index.js

```diff
@@ -297,6 +297,10 @@ function getContextDB(key, opts) {
       toUseAppId = getDevelopmentAppID(appId)
       break
   }
+
+  //If the appId is specified, override context
+  toUseAppId = opts?.appId ? opts?.appId : toUseAppId
```

Comment:
I think this might have the potential to cause side effects where a subsequent call to `getAppDB` would now be using the supplied app id instead of the current, dirtying the current context.  It might be safer to wrap the implementation of `updateAppPackage` with `doInAppContext`, which would create a new context stack for the logic being wrapped, this would also make sure the correct tenantId is set according to the appId supplied.

Reply:
best to check with @mike12345567

Reply:
Something like
```js
const updateAppPackage = async (appPackage: any, appId: any) => {
   doInAppContext(appId, async () => {

   }
})

```

Reply:
Thanks for looking at this. Yes, I see what you're getting at. I've pushed an update that follows the approach that you suggested. I retested the areas that make calls to `updateAppPackage` and the behaviour appears to be as expected.

Reply:
Sorry just seeing this, yes wrapping the whole API endpoint in another app context is the safest option.

One question I have is how there is a mis-match in app IDs to begin with, does the request contain both the production and development app IDs?

Reply:
Hey, thanks for taking a look! Yes, in this instance there was a mis-match. When updating the icon for a published app in the app list, it resolved the context as the prod application. In the overview screen, the context resolves itself as the dev application. 
In order to mirror the same behaviour as the app list I needed to explicitly set the context.

Reply:
ah I see, it might be worth seeing why the overview screen resolves to the dev app, I'm assuming either the `x-budibase-app-id` header isn't being set correctly and its falling back to the cookie app ID. Sometimes app IDs within the request get ignored for the context management as well, but the easiest way to get around this is to always declare the correct app ID in the `x-budibase-app-id` header.